### PR TITLE
Remember GUI column sort and separate hotkey settings from emulated input settings

### LINKED ISSUE #1214: Make Ryujinx games list "sort by" method persistent
Currently Ryujinx has no memory of how you sorted your games list the last time you were using the application.
It would improve the user experience to be able to sort games a particular way, and then have that setting remembered the next time you close and open Ryujinx again.

### PR DESCRIPTION
**Pull request by Xpl0itR**
Fix #1214

### CODE REVIEW

File: Ryujinx.Common/Configuration/ConfigurationState.cs [hunk 1]
```diff
@@ -398,11 +428,17 @@ public void LoadDefault()
...
             Ui.GuiColumns.PathColumn.Value         = true;
+            Ui.ColumnSort.SortColumnId.Value       = 0;
+            Ui.ColumnSort.SortAscending.Value      = false;
             Ui.GameDirs.Value                      = new List<string>();
             Ui.EnableCustomTheme.Value             = false;
             Ui.CustomThemePath.Value               = "";
             Hid.EnableKeyboard.Value               = false;
-
+            
+            Hid.Hotkeys.Value     = new KeyboardHotkeys
```
Comment (AcK77) [inline Ryujinx.Common/Configuration/ConfigurationState.cs:438]: nit:
```suggestion
            Hid.Hotkeys.Value = new KeyboardHotkeys
```

File: Ryujinx/_schema.json [hunk 2]
```diff
@@ -539,6 +539,16 @@
...
+    "column_sort": {
+      "$id": "#/properties/column_sort",
+      "type": "array",
+      "title": "Used to configure column sort settings in the GUI",
+      "description": "Used to configure column sort settings in the GUI",
+      "default": {
+        "sort_column_id": 0,
+        "sort_ascending": false
+      }
+    },
```
Comment (jduncanator) [inline Ryujinx/_schema.json:551]: This isn't valid. Child properties should be implemented the same as the `hotkeys` entry.

**Comment by Xpl0itR** [comment]
Fixed the multiple issues in the schema

File: Ryujinx.Common/Configuration/ConfigurationState.cs [hunk 3]
```diff
@@ -398,11 +428,17 @@ public void LoadDefault()
...
             Ui.GameDirs.Value                      = new List<string>();
             Ui.EnableCustomTheme.Value             = false;
             Ui.CustomThemePath.Value               = "";
             Hid.EnableKeyboard.Value               = false;
-
+            
+            Hid.Hotkeys.Value = new KeyboardHotkeys
+            {
+                ToggleVsync = Key.Tab
+            };
```
Comment (AcK77) [inline Ryujinx.Common/Configuration/ConfigurationState.cs:441]: Add a new empty line after.

File: Ryujinx/Ui/KeyboardController.cs [hunk 4]
```diff
@@ -91,20 +92,19 @@ public ControllerKeys GetButtons()
...
 
-        public HotkeyButtons GetHotkeyButtons()
+        public static HotkeyButtons GetHotkeyButtons(KeyboardState keyboard)
         {
-            KeyboardState keyboard = GetKeyboardState(_config.Index);
-
             HotkeyButtons buttons = 0;
 
-            if (keyboard[(Key)_config.Hotkeys.ToggleVsync]) buttons |= HotkeyButtons.ToggleVSync;
+            if (keyboard[(Key)ConfigurationState.Instance.Hid.Hotkeys.Value.ToggleVsync])
```
Comment (AcK77) [inline Ryujinx/Ui/KeyboardController.cs:99]: Maybe add brackets here?

File: Ryujinx/Ui/MainWindow.cs [hunk 5]
```diff
@@ -699,6 +733,15 @@ private void FavToggle_Toggled(object sender, ToggledArgs args)
             });
         }
 
+        private void Column_Clicked(object sender, EventArgs args)
+        {
+            TreeViewColumn column = (TreeViewColumn)sender;
+
+            ConfigurationState.Instance.Ui.ColumnSort.SortColumnId.Value  = column.SortColumnId;
+            ConfigurationState.Instance.Ui.ColumnSort.SortAscending.Value = column.SortOrder == SortType.Ascending;
+            SaveConfig();
```
Comment (AcK77) [inline Ryujinx/Ui/MainWindow.cs:742]: Maybe add a new empty line before?

**Comment by EmulationFanatic** [comment]
The sorting is indeed remembered! Excellent work.
However, as mentioned on Discord, a GUI bug not (yet) addressed by this PR is that when the list of games has been loaded, the scroll bar is not completely at the top, leaving one to several games above the displayed top-of-the-list game after opening Ryujinx. Xpl0itR has stated he knows the cause of this and will resolve the issue.

See GIF below (ignore green color gif encoding defect) showing Alien Escape at the top, then closing Ryujinx, reopening it, and having to scroll up to put Alien Escape back at the top after the list has loaded:
GUInofix

**Comment by EmulationFanatic** [comment]
Tested again after the last two commits, and I am a huge fan of the GUI game list load process now. Awesome work!

**Pull request merged**